### PR TITLE
feat(helm chart): prepare helm chart for release 4.1.0

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
### Changes
- Updates the dependent charts with their respective latest releases.
- Updates the relevant README

### CODE REVIEW

**Review by niladrih — CHANGES_REQUESTED** [review]
You'd have to add monitoring here
Instead of 'engines', you can put it under 'utils'?
WDYT @tiagolobocastro @avishnu @dsharma-dc @w3aman @Abhinandan-Purkait @abhilashshetty04 ?

**Comment by Abhinandan-Purkait** [comment]
@w3aman Please add the monitoring in this PR.

**Comment by Abhinandan-Purkait** [comment]
> You'd have to add monitoring here Instead of 'engines', you can put it under 'utils'? WDYT @tiagolobocastro @avishnu @dsharma-dc @w3aman @Abhinandan-Purkait @abhilashshetty04 ?

How about just monitoring?

**Comment by niladrih** [comment]
@Abhinandan-Purkait please also incorporate my commit from https://github.com/openebs/openebs/pull/3747.

**Comment by niladrih** [comment]
> > You'd have to add monitoring here Instead of 'engines', you can put it under 'utils'? WDYT @tiagolobocastro @avishnu @dsharma-dc @w3aman @Abhinandan-Purkait @abhilashshetty04 ?
> 
> How about just monitoring? In the values root.

Yep, works for me. `monitoring.enabled`.

**Comment by Abhinandan-Purkait** [comment]
> @Abhinandan-Purkait please also incorporate my commit from #3747.

Done. Please check.

**Review by niladrih — APPROVED** [review]
LGTM!

File: charts/values.yaml [hunk 1]
```diff
@@ -39,12 +39,21 @@ mayastor:
     node:
       initContainers:
         enabled: true
-
+  etcd:
```
Comment (tiagolobocastro) [inline charts/values.yaml:42]: above on line 35, change to 2.7.0 ?

Comment (Abhinandan-Purkait) [reply charts/values.yaml:42]: Done.

**Review by tiagolobocastro — APPROVED** [review]

File: charts/values.yaml [hunk 2]
```diff
@@ -32,19 +32,28 @@ lvm-localpv:
...
     enabled: false
   crds:
     enabled: false
 
+# Refer to https://github.com/openebs/monitoring/blob/HEAD/deploy/charts/values.yaml for complete set of values.
+monitoring:
+  # -- Enable monitoring stack for openebs engines
+  enabled: false
+  localpv-provisioner:
+    enabled: false
```
Comment (niladrih) [inline charts/values.yaml:55]:
```suggestion
```

**Review by niladrih — CHANGES_REQUESTED** [review]

File: charts/README.md [hunk 3]
```diff
@@ -108,19 +108,18 @@ helm delete `<RELEASE NAME>` -n `<RELEASE NAMESPACE>`
 | localpv-provisioner.rbac.create | bool | `true` |  |
 | lvm-localpv.crds.csi.volumeSnapshots.enabled | bool | `false` |  |
 | lvm-localpv.crds.lvmLocalPv.enabled | bool | `true` |  |
-| lvm-localpv.enabled | bool | `true` |  |
 | mayastor.crds.csi.volumeSnapshots.enabled | bool | `false` |  |
 | mayastor.csi.node.initContainers.enabled | bool | `true` |  |
-| mayastor.enabled | bool | `true` |  |
+| mayastor.etcd.clusterDomain | string | `"cluster.local"` | Kubernetes Cluster Domain |
 | mayastor.localpv-provisioner.enabled | bool | `false` |  |
+| monitoring.enabled | bool | `false` | Enable monitoring stack for openebs engines |
```
Comment (niladrih) [inline charts/README.md:115]:
```suggestion
```

File: charts/README.md [hunk 4]
```diff
@@ -72,7 +72,7 @@ To view the chart and get the following output.
 helm ls -n openebs 
 
 NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
-openebs openebs         1               2024-03-25 09:13:00.903321318 +0000 UTC deployed        openebs-4.0.1   4.0.1
+openebs openebs         1               2024-03-25 09:13:00.903321318 +0000 UTC deployed        openebs-4.1.0   4.1.0
```
Comment (niladrih) [inline charts/README.md:75]:
```suggestion
openebs openebs         1               2024-07-07 09:13:00.903321318 +0000 UTC deployed        openebs-4.1.0   4.1.0
```

**Review by niladrih — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
> Please squash all of the commits, if possible.

Isn't squash and merge enough here?

**Pull request merged**